surefire-api/src/main/java/org/apache/maven/surefire/api/util/DefaultRunOrderCalculator.java

```diff
@@ -181,14 +143,55 @@ public void addTestToOrders( String className, LinkedHashMap<String, List<String
         }
     }
 
+    public TestListResolver getTestListResolver()
+    {
+        String orderParam = parseTestOrder( System.getProperty( "test" ) );
+        if ( orderParam == null  )
+        {
+            throw new IllegalStateException( "Please set system property -Dtest to use fixed order"  );
+        }
+        List<String> list = Arrays.asList( orderParam.split( "," ) );
+        return new TestListResolver( list );
+    }
+
+    public String[] getClassAndMethod( String request )
+    {
+        String[] classAndMethod = { request, request };
+        if ( request.contains( "(" ) )
+        {
+            String[] nameSplit1 = request.split( "\\(" );
+            classAndMethod[0] = nameSplit1[1].substring( 0, nameSplit1[1].length() - 1 );
+            classAndMethod[1] = nameSplit1[0];
+        }
+        return classAndMethod;
+    }
+
+    private RunOrder getRunOrderMethod()
+    {
+        if ( runOrder.length == 0 )
+        {
+            throw new IllegalStateException( "Expected runOrder to be testorder." );
+        }
+        List<RunOrder> runOrderList = new ArrayList<RunOrder>( Arrays.asList( runOrder ) );
```

Comment:
Instead of creating a new list object just to call contains, you should be able to use Streams or other more efficient ways of checking contains. E.g.,
https://stackoverflow.com/a/1128728

Reply:
I get this error:
[ERROR] /mnt/batch/tasks/workitems/SUA_tmp_r0_1M7d21h27m24s/job-1/Task0/wd/maven-surefire/surefire-api/src/main/java/org/apache/maven/surefire/api/util/DefaultRunOrderCalculator.java:[171,65] error: lambda expressions are not supported in -source 1.7
[ERROR]   (use -source 8 or higher to enable lambda expressions)

Reply:
Is the error from using Streams?

What happens when you use ```Arrays.asList(yourArray).contains(yourValue)``` instead?